File: FRONTEND/src/app/data/event/event.model.ts
import { EventHasPosition } from "../eventHasPosition/eventHasPosition.model";
import { Recruiter } from "../recruiter/recruiter.model";

export class Event {
	eventId?: string;
	eventName?: string;
	recruiterId?: string;
	recruiter?: Recruiter;
	description?: string;
	imageURL?: string;
	applyPriority?: number;
	place?: string;
	startDateTime?: Date;
	endDateTime?: Date;
	maxParticipants?: number;
	isDeleted?: boolean;
	eventHasPositions?: EventHasPosition[];
}

export class EventFilter {
	search?: string;
	fromDate?: Date;
	toDate?: Date;
	fromMaxParticipants?: number;
	toMaxParticipants?: number;
}

File: BackEnd/Service/Models/EventModel.cs
namespace Service.Models;

public class EventModel
{
    public Guid EventId { get; set; }

    public string EventName { get; set; } 

    public Guid RecruiterId { get; set; }

    public RecruiterModel Recruiter { get; set; }

    public string? Description { get; set; }

    public string Place { get; set; } 

    public string? ImageURL { get; set; }

    public DateTime StartDateTime { get; set; }

    public DateTime EndDateTime { get; set; }

    public int MaxParticipants { get; set; }

    public bool IsDeleted { get; set; } = false;
    public List<EventHasPositionModel> EventHasPositions { get; set; }
}